webservices/server-integration/src/main/java/org/jboss/as/webservices/deployers/AspectDeploymentProcessor.java

```diff
@@ -112,4 +124,33 @@ private static boolean isWebServiceDeployment(final DeploymentUnit unit) {
         return unit.getAttachment(WSAttachmentKeys.DEPLOYMENT_KEY) != null;
     }
 
+
+    // Fix for WFLY-12135. We need to let JBossWS know that there's the only listeners available are HTTPS ones.
+    private boolean isHttpsOnly(final DeploymentPhaseContext phaseContext, final DeploymentUnit unit) {
+        CapabilityServiceSupport serviceSupport = unit.getAttachment(Attachments.CAPABILITY_SERVICE_SUPPORT);
+        ServiceName undertowServiceName = serviceSupport.getCapabilityServiceName("org.wildfly.undertow");
+        ServiceController<?> undertowServiceController = phaseContext.getServiceRegistry().getService(undertowServiceName);
+        UndertowService undertowService = (UndertowService) undertowServiceController.getValue();
+
+        Server defaultServer = null;
+
+        for (Server server : undertowService.getServers()) {
+            if (server.getName().equals("default-server")) {
```

Comment:
We shouldn't use a magic value. UndertowService exposes a getDefaultServer() which seems like it should provide the correct value.